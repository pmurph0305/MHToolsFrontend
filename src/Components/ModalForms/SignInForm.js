import React from "react";
import PropTypes from "prop-types";
import "./SignInForm.scss";

/** SignInForm form - returns a component that renders a signin form
 * @callback onSubmitForm called when the user clicks the submit button
 * @callback onToggleModal called when the user clicks the close (X) button on the modal.
 */
const SignInForm = ({ onSubmitForm, onToggleModal }) => {
  return (
    <div className="form-container">
      <form className="form-modal" onSubmit={onSubmitForm}>
        <div className="modal-close" onClick={onToggleModal}>
          &times;
        </div>
        <div className="form-flex-container">
          <h1 className="form-title">Sign in</h1>
          <div className="form-entries-container">
            <div className="form-input-container">
              <label className="form-label" htmlFor="email">
                Email:
              </label>
              <input
                autoComplete="email"
                className="form-input"
                type="text"
                placeholder="Example@gmail.com"
                name="email"
                id="email"
                required
              />
            </div>
            <div className="form-input-container">
              <label className="form-label" htmlFor="password">
                Password:
              </label>
              <input
                autoComplete="current-password"
                className="form-input"
                type="password"
                placeholder="password"
                name="password"
                id="password"
                required
              />
            </div>
            <input
              type="hidden"
              id="hiddenFormEl"
              name="hidden"
              value="hiddenValue"
            />
            <input className="form-submit" type="submit" value="Sign in" />
          </div>
        </div>
      </form>
    </div>
  );
};

SignInForm.propTypes = {
  onSubmitForm: PropTypes.func.isRequired,
  onToggleModal: PropTypes.func.isRequired
};

export default SignInForm;
