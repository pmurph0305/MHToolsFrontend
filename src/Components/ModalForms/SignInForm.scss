@import "../../styles/shared.scss";

.form-container {
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;

  .form-title {
    font-size: 1.5rem;
    margin: 0.5rem auto;
  }

  .form-modal {
    background-color: $primary-background;
    border-radius: 0.5rem;
    border-style: solid;
    border-width: 1px;
    border-color: $secondary-color;
    margin: 2rem auto;
    text-align: center;
    /* Add Media-queries. */
    width: 75%;
    max-width: 35rem;
    box-shadow: 0px 0px 2px 0px $secondary-color;
    padding-bottom: 2rem;
  }

  .form-flex-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .modal-close {
    float: right !important;
    font-size: 2.5rem;
    padding-right: 0.75rem;
    font-style: bold;
    cursor: pointer;
  }

  .modal-close:hover {
    color: $secondary-color;
    cursor: pointer;
  }

  .form-entries-container {
    margin: auto;
    display: block;
    height: 100%;
  }

  .form-input-container {
    font-size: 1.25rem;
    font-weight: 600;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
    text-align: left;
    padding-left: 0.5em;
  }

  label {
    line-height: 1.25rem;
    padding-right: 1rem;
    text-align: left;
  }

  .form-submit {
    @extend %ButtonBase;
    display: block;
    margin: auto;
    padding: 0.5rem 1rem;
    margin-top: 0.5rem;
  }

  .form-input {
    background-color: $secondary-background;
    padding: 0.25em 0.5em;
    width: 100%;
  }
}
