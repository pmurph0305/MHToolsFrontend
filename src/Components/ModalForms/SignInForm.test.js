import { shallow } from "enzyme";
import React from "react";
import SignInForm from "./SignInForm";

describe("AddSkill tests", () => {
  let mockOnSubmit = jest.fn(val => val);
  let mockOnToggleModal = jest.fn(val => val);
  let mockProps = {
    onSubmitForm: mockOnSubmit,
    onToggleModal: mockOnToggleModal
  };
  const wrapper = shallow(<SignInForm {...mockProps} />);

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onsubmit", () => {
    wrapper.find("form").simulate("submit");
    expect(mockOnSubmit).toHaveBeenCalledTimes(1);
  });

  it("calls toggle properly", () => {
    wrapper.find('[className="modal-close"]').simulate("click");
    expect(mockOnToggleModal).toHaveBeenCalledTimes(1);
  });
});
