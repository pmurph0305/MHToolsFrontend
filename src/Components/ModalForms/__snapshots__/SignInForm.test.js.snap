// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm tests matches snapshot 1`] = `ShallowWrapper {}`;
